MSOE.MediaComplete.Lib/Dependency.cs

```diff
@@ -0,0 +1,66 @@
+﻿using Autofac;
+using Autofac.Core;
+using MSOE.MediaComplete.Lib.Background;
+using MSOE.MediaComplete.Lib.Files;
+using MSOE.MediaComplete.Lib.Import;
+using MSOE.MediaComplete.Lib.Metadata;
+using MSOE.MediaComplete.Lib.Playing;
+using MSOE.MediaComplete.Lib.Playlists;
+using MSOE.MediaComplete.Lib.Sorting;
+
+namespace MSOE.MediaComplete.Lib
+{
+    public static class Dependency
+    {
+        private static IContainer _afContainer;
+        
+        public static async void BuildAsync()
+        {
+            var builder = new ContainerBuilder();
+            var fileManager = FileManager.Instance;
+            fileManager.Initialize(SettingWrapper.MusicDir);
+            builder.RegisterInstance(fileManager).ExternallyOwned().As<IFileManager>();
+            builder.RegisterInstance(StatusBarHandler.Instance);
+            builder.RegisterType<FfmpegAudioReader>().As<IAudioReader>();
+            builder.RegisterType<DoresoIdentifier>().As<IAudioIdentifier>();
+            builder.RegisterInstance(await SpotifyMetadataRetriever.GetInstanceAsync()).ExternallyOwned().As<IMetadataRetriever>();
+            builder.RegisterType<Identifier>().WithParameters(new[]
+            {
+                new ResolvedParameter((pi, c) => pi.ParameterType == typeof(IAudioReader), (pi, c) => c.Resolve<IAudioReader>()),
+                new ResolvedParameter((pi, c) => pi.ParameterType == typeof(IAudioIdentifier), (pi, c) => c.Resolve<IAudioIdentifier>()),
+                new ResolvedParameter((pi, c) => pi.ParameterType == typeof(IMetadataRetriever), (pi, c) => c.Resolve<IMetadataRetriever>()),
+                new ResolvedParameter((pi, c) => pi.ParameterType == typeof(IFileManager), (pi, c) => c.Resolve<IFileManager>())
+            });
+            builder.RegisterType<Sorter>().WithParameters(new[]
+            {
+                new ResolvedParameter((pi, c) => pi.ParameterType == typeof(IFileManager), (pi, c) => c.Resolve<IFileManager>())
+            });
+            builder.RegisterType<Importer>().WithParameters(new[]
+            {
+                new ResolvedParameter((pi, c) => pi.ParameterType == typeof(IFileManager), (pi, c) => c.Resolve<IFileManager>())
+            });
+            builder.RegisterType<NAudioWrapper>().As<INAudioWrapper>();
+            var polling = new Polling();
+            builder.RegisterInstance(polling).As<IPolling>();
+
+            builder.RegisterType<PlaylistServiceImpl>().WithParameters(new[]
+            {
+                new ResolvedParameter((pi, c) => pi.ParameterType == typeof(IFileManager), (pi, c) => c.Resolve<IFileManager>())
+            });
+            var queue = new Queue();
+            builder.RegisterInstance(queue).As<IQueue>();
+            _afContainer = builder.Build();
+        }
+
+        public static T Resolve<T>()
```

Comment:
Importer, Sorter, and Identified all use scope. The non singleton dependencies that use resolve instead of a scope is NAudioWrapper, which was created by doing (new wrapper()) before. I don't agree that this API skews towards directly resolving instead of using a scope. I think both options are presented and its the developers choice on which to use.

Lets, for a moment look at where I deleted this from. In the MainWindow.xaml.cs, the builder was defined and built, and BeginLifetimeScope() was called on the container. You could have used Resolve<T> instead, but you didn't since it was not the right way to use it. I exposed the same functions that existed before when you had the option of using it one way and you didn't.